### PR TITLE
fix: adding observer for finace and recent report

### PR DESCRIPTION
**Pull request by anagarwa**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--icicidirect--aemsites.hlx.live/research/equity
- After: https://observer-add--icicidirect--aemsites.hlx.live/research/equity

### CODE REVIEW

File: blocks/recentreports/recentreports.js [hunk 1]
```diff
@@ -215,29 +219,23 @@ function addDiscoverLink(discoverMoreDiv, block) {
...
+      const element = block.querySelector('.carousel-wrapper');
+      handleNoResults(element);
+    } else {
+      renderRecentReportsCards(
+        recentReportsDataArray,
+        carouselItems,
+        title,
+        block?.dataset?.maxLimit,
+      );
+      observe(block, loadCarousel, carouselItems);
```
Comment (JiangLong2019) [inline blocks/recentreports/recentreports.js:236]: Do we need this `observe` to handle carousel rendering?

Comment (anagarwa) [reply blocks/recentreports/recentreports.js:236]: @JiangLong2019 carousel is created here https://github.com/aemsites/icicidirect/pull/289/files#diff-e7ff8117e0e6827be45cbee4d72b0300fb5d7d7ac8bc8f6cbcfa26c10f7eac62R295, the dyamic data is added in the observe

Comment (JiangLong2019) [reply blocks/recentreports/recentreports.js:236]: @anagarwa there are two `observe`. The first one[0] we use to call API when the block comes to viewpoint. In [0], we are calling another `observe`[1] which loads the carousel. Do you think we need the second `observe` to double check the viewpoint?

[0] - https://github.com/aemsites/icicidirect/pull/289/files#diff-e7ff8117e0e6827be45cbee4d72b0300fb5d7d7ac8bc8f6cbcfa26c10f7eac62R296
[1] - https://github.com/aemsites/icicidirect/pull/289/files#diff-e7ff8117e0e6827be45cbee4d72b0300fb5d7d7ac8bc8f6cbcfa26c10f7eac62R236

Comment (anagarwa) [reply blocks/recentreports/recentreports.js]: @JiangLong2019  yes, i missed that updated the code

Comment (JiangLong2019) [reply blocks/recentreports/recentreports.js]: Thanks

**Review by JiangLong2019 — APPROVED** [review]

**Pull request merged**